osu.Server.Queues.ScoreStatisticsProcessor/Helpers/ModHelper.cs

```diff
@@ -0,0 +1,28 @@
+﻿using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu.Mods;
+
+namespace osu.Server.Queues.ScoreStatisticsProcessor.Helpers
+{
+    public static class ModHelper
+    {
+        public static bool IsDifficultyReductionMod(this Mod mod)
+        {
+            switch (mod)
+            {
+                case ModEasy:
+                case ModNoFail:
+                case ModHalfTime:
+                case OsuModSpunOut:
+                    return true;
+
+                // Difficulty adjust can be used to decrease the difficulty of the map,
+                // so let's consider it difficulty reducing for now.
+                case ModDifficultyAdjust difficultyAdjust:
+                    return !difficultyAdjust.UsesDefaultConfiguration;
```

Comment:
If we're going there we probably need to exclude a lot more...

- relax
- autopilot
- daycore

etc. etc.

Tempted to just say we should be switching this to something like `mod.Type == ModType.DifficultyReduction || mod.Type == ModType.Automation` at that point. Maybe `ModType.Fun` too, not sure.

Reply:
I made it check for `ModType.DifficultyReduction` and `ModType.Automation`, most of `ModType.Fun` except a couple, then also exclude some other mods (DA and TP). I think this should cover it pretty well.

Reply:
I'm still not sure this goes far enough. Most conversion mods (random, invert, mirror...) should probably be disallowed too.

In fact, one point to this is that because this is looking up difficulty attributes from database (until we get realtime difficulty working), _no_ new mods should be allowed for star rating medals (unless one makes an executive decision that they don't affect anything), because we _don't have_ accurate star ratings with arbitrary mods. Only the combinations that stable allows should be supported.

So for now we probably want to swap this blocklist to an allowlist instead that matches what stable permits exactly.

Reply:
Implemented as an allow-list for now, can be adjusted whenever SR has support for the other mods.

Reply:
I've reformulated this slightly with the new `Ranked` property in d25b6fddc48e8f9e654fac912064e76dcfe51d05 (test coverage at 532ec32ad6baee2cf88173283828173b53c516f0). Not sure how this is going to hold up long-term, but let's start with this and see where this goes.